api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -411,24 +412,26 @@ public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBilli
       fireCloudService.updateBillingAccountAsService(
           workspace.getWorkspaceNamespace(), newBillingAccountName);
     } else {
-      try {
-        Optional<Boolean> isOpenMaybe =
-            Optional.ofNullable(
-                cloudBillingClient.getBillingAccount(newBillingAccountName).getOpen());
-        boolean isOpen = isOpenMaybe.orElse(false);
-        if (!isOpen) {
-          throw new BadRequestException(
-              "Provided billing account is closed. Please provide an open account.");
-        }
-        fireCloudService.updateBillingAccount(
-            workspace.getWorkspaceNamespace(), newBillingAccountName);
-      } catch (IOException e) {
-        throw new ServerErrorException("Could not fetch user provided billing account.", e);
+      fireCloudService.updateBillingAccount(
+          workspace.getWorkspaceNamespace(), newBillingAccountName);
+    }
+
+    try {
+      ProjectBillingInfo projectBillingInfo =
+          cloudBillingClient.pollUntilBillingAccountLinked(
+              newBillingAccountName, workspace.getGoogleProject());
+      if (!projectBillingInfo.getBillingEnabled()) {
+        throw new BadRequestException(
+            "Provided billing account is closed. Please provide an open account.");
       }
+    } catch (IOException | InterruptedException e) {
+      throw new ServerErrorException(
+          String.format(
+              "Error when poll billing account %s for project %s",
```

Comment:
This string is meant to potentially be shown to the user, so I would not include internal implementation details like this. Something like : Timed out while verifying billing account update.